lib/trino-orc/src/main/java/io/trino/orc/metadata/statistics/StringStatisticsBuilder.java

```diff
@@ -158,9 +173,19 @@ public ColumnStatistics buildColumnStatistics()
         return stringStatisticsBuilder.buildStringStatistics();
     }
 
-    private Slice dropStringMinMaxIfNecessary(Slice minOrMax)
+    private Slice dropStringMinMaxIfNecessary(Slice minOrMax, boolean isMin)
     {
         if (minOrMax == null || minOrMax.length() > stringStatisticsLimitInBytes) {
+            if (minOrMax != null && shouldCompactMinMax) {
+                if (isMin) {
+                    byte[] min = StringCompactor.computeMin(minOrMax.getBytes(), stringStatisticsLimitInBytes);
+                    return Slices.wrappedBuffer(min);
```

Comment:
The method name `dropStringMinMaxIfNecessary` no longer applies.

`compactStringMinMaxIfNecessary` ?

Reply:
I will even go with computeStringMinMax